src/TunNetCom.SilkRoadErp.Sales.Api/Features/Products/CreateProduct/CreateProductEndpoint.cs

```diff
@@ -0,0 +1,43 @@
+﻿using TunNetCom.SilkRoadErp.Sales.Contracts.Products;
+
+namespace TunNetCom.SilkRoadErp.Sales.Api.Features.Products.CreateProduct
+{
+    public class CreateProductEndpoint : ICarterModule
+    {
+        public void AddRoutes(IEndpointRouteBuilder app)
+        {
+            _ = app.MapPost(
+             "/Products",
+             async Task<Results<Created<CreateProductRequest>, BadRequest<List<IError>>>>
```

Comment:
async Task<Results<Created<CreateProductRequest>, ValidationProblem>>

Reply:
the result of validationProblem is  int not string thats why i didn't use it  and type of refe is string